builtin/logical/aws/path_static_roles.go

```diff
@@ -0,0 +1,330 @@
+					Type:        framework.TypeString,
+					Description: descUsername,
+				},
+				paramRotationPeriod: {
+					Type:        framework.TypeDurationSecond,
+					Description: descRotationPeriod,
+				},
+			},
+		}},
+	}
+
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s", pathStaticRole, framework.GenericNameWithAtRegex(paramRoleName)),
+		Fields: map[string]*framework.FieldSchema{
+			paramRoleName: {
+				Type:        framework.TypeString,
+				Description: descRoleName,
+			},
+			paramUsername: {
+				Type:        framework.TypeString,
+				Description: descUsername,
+			},
+			paramRotationPeriod: {
+				Type:        framework.TypeDurationSecond,
+				Description: descRotationPeriod,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:  b.pathStaticRolesRead,
+				Responses: roleResponse,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesWrite,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses:                   roleResponse,
+			},
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesDelete,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses: map[int][]framework.Response{
+					http.StatusNoContent: {{
+						Description: http.StatusText(http.StatusNoContent),
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathStaticRolesHelpSyn,
+		HelpDescription: pathStaticRolesHelpDesc,
+	}
+}
+
+func (b *backend) pathStaticRolesRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing %q parameter", paramRoleName)
+	}
+
+	b.roleMutex.RLock()
+	defer b.roleMutex.RUnlock()
+
+	entry, err := req.Storage.Get(ctx, formatRoleStoragePath(roleName.(string)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read configuration for static role %q: %w", roleName, err)
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var config staticRoleEntry
+	if err := entry.DecodeJSON(&config); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for static role %q: %w", roleName, err)
+	}
+
+	return &logical.Response{
+		Data: formatResponse(config),
+	}, nil
+}
+
+func (b *backend) pathStaticRolesWrite(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// Create & validate config from request parameters
+	config := staticRoleEntry{}
+	isCreate := req.Operation == logical.CreateOperation
+
+	if rawRoleName, ok := data.GetOk(paramRoleName); ok {
+		config.Name = rawRoleName.(string)
+
+		if err := b.validateRoleName(config.Name); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing %q parameter", paramRoleName)
```

Comment:
I'm pretty sure returning an error will result in a 500. Suggest to use `return logical.ErrorResponse("message"), nil` for any client errors so that they come back as a 400. Documented [here](https://hashicorp.atlassian.net/wiki/spaces/VAULT/pages/1190494373/Path+Definitions).

Reply:
done for validation!